dask/array/routines.py

```diff
@@ -312,25 +312,56 @@ def _inner_apply_along_axis(arr,
     )
 
 
-@wraps(np.apply_along_axis)
-def apply_along_axis(func1d, axis, arr, *args, **kwargs):
+def apply_along_axis(func1d, axis, arr, dtype=None, shape=None, *args, **kwargs):
```

Comment:
So the one concern that comes to mind when inserting these keywords here is they will now also pick up what use to be the first two user arguments. We can fix that by actually extracting these from `kwargs` instead. Thus preserving the API and providing this functionality.

Reply:
ok python 101 reminder: no repeated keyword-arguments, so the ambiguity I referenced above isn't possible. That being said, picking essential keyword-arguments out of `kwargs` feels a little hacky.

Reply:
Certainly understand that. If we were using Python 3 only, we could use keyword-only arguments, which would be better.

Extracting the keyword from `kwargs` seems like the least bad option to me. Though maybe someone else has thoughts here.

cc @mrocklin @jcrist

Reply:
`dtype` and `shape` are now extracted from `**kwargs` in [4a112d0](https://github.com/dask/dask/pull/3742/commits/4a112d09e13f1d97db9b1e19f3ffac40bb3877b7)